docs/ref/forms/widgets.txt

```diff
@@ -569,6 +569,30 @@ These widgets make use of the HTML elements ``input`` and ``textarea``.
     * ``template_name``: ``'django/forms/widgets/search.html'``
     * Renders as: ``<input type="search" ...>``
 
+``TelInput``
+~~~~~~~~~~~~~~
+
+.. versionadded:: 5.2
+
+.. class:: TelInput
+
+    * ``input_type``: ``'tel'``
+    * ``template_name``: ``'django/forms/widgets/tel.html'``
+    * Renders as: ``<input type="tel" ...>``
+
+    The input value is not automatically validated to a particular format
+    before the form can be submitted, because formats for telephone numbers
+    vary so much around the world.
+
+    Client-side validations can be added by setting ``pattern``, ``minlength``
+    and ``maxlength`` keys on the :attr:`Widget.attrs` argument.
+
+    Server-side validations can be achieved through the use of third-party
+    packages, such as `django-phonenumber-field`_.
```

Comment:
Server-side validation isn't really related to the widget, but the field, so I’d make this an “also” section. And let’s mention the simpler options built in to Django first, probably many projects only need take phone numbers in one format.

```suggestion
    Additionally, you can add server-side validation to your form field with a
    validator like :class:`~django.core.validators.RegexValidator` or
    third-party packages, such as `django-phonenumber-field`_.
```

Reply:
Good call! :+1: